LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_event_basic.https.window-expected.txt

```diff
@@ -1,5 +1,3 @@
 
-Harness Error (TIMEOUT), message = null
-
-TIMEOUT cookieStore fires change event for cookie set by cookieStore.set() Test timed out
+FAIL cookieStore fires change event for cookie set by cookieStore.set() assert_equals: expected [object "[object Object]"] but got [object "[object Object]"]
```

Comment:
Would be nice to pass this "basic" test though.. :)

Reply:
Yeah, that's true. I'll look into this and try and fix this in this patch itself

Reply:
I'm a bit confused about this test. The line that is failing is `assert_equals(event.changed, event.changed);` And it's saying that `assert_equals: expected [object "[object Object]"] but got [object "[object Object]"]`. But aren't these the same variables that it's comparing?

Reply:
If you comment out those 2 lines:
```
  assert_equals(event.changed, event.changed);
  assert_equals(event.deleted, event.deleted);
```

Is the test passing? If so, we can address I think you just need to add [CachedAttribute] in the IDL for `changed` & `deleted`.

`changed` & `deleted` are likely marked as [SameObject] in the idl but our implementation is currently slightly wrong and doesn't necessarily return the same objects.

Reply:
Yes, if we comment out those lines, then the test passes. And yes, `changed` and `deleted` are marked as [SameObject]. What exactly does that mean? And what is the part of the implementation that is lightly wrong. Maybe I can try fixing it?

Reply:
I told you how to fix it in my comment already.

Reply:
Ah ok. I thought we needed [SameAttribute] eventually and there was some other implementation mistake and [CachedAttribute] would be a temporary fix for now. But I can change it to [CachedAttribute]. What does Cached Attribute mean here?

Reply:
[CachedAttribute] is one of the way to implement the SameObject behavior, it is the simplest one. A new data member is added to the JS wrapper to cache the value returned by the attribute the first time it is accessed. Then on follow-up calls, it will return this cached value.

You want to use `[CachedAttribute, SameObject]` in the IDL in your case. I believe it will fix your test and is a proper fix.

Reply:
Ok that makes sense. It should be fixed now!